### ISSUE TITLE
Work area issues .. could it be bigger?

### ISSUE THREAD

**Issue by dobkeratops** [ImageMonkey/imagemonkey-core#290]
Would it be possible to tweak how the image work area is calculated..
Sometimes it is actually *smaller* than in the browser view. Far too much screen space goes on padding.
This is especially bad on the iPad which is a shame because annotating with the pen is amazingly fast.. but 
![BB712BBC-7908-4B4D-A94F-C149E20F669D](https://user-images.githubusercontent.com/1120754/115059736-dc3f5b00-9ede-11eb-88fb-dac141df829d.jpeg)
we can’t do it accurately with just 25% of its smaller screen
Suggestions -
* make the too banner much smaller - even get rid of the bug “annotate all..” text - you’ve got the selector in the label list, and the overlay indicator in the bottom right (superior because it never scrolls out of view)
* Make the “properties” hideable, or move it to the left,under the the main label list (think of how photoshop deals with palettes). 

Example of worst case tiny work area :
![3EA1DCBA-0BEC-4005-84A1-5752FA1D0E71](https://user-images.githubusercontent.com/1120754/115059598-ab5f2600-9ede-11eb-8809-75edc18065a1.jpeg)

**Comment by dobkeratops** [comment]
Going back to the ipad... what I’d tended to do is add labels on a pc (easier to type) but there’s now a neat handwriting text entry feature in iOS, which makes pure iPad use a bit better. (One caveat it keeps turning the entry into uppercase first labels.. I didn’t want to further pollute the data with those.. so I’ve stuck to alternating )

**Comment by bbernhard** [comment]
Yeah, that's the problem of those responsive design frameworks :/ But I think there are some options to change the layout depending on the device one is using. Do you use the properties on the ipad or should we maybe hide it completely from the user interface?

**Comment by dobkeratops** [comment]
Completely hiding properties if the window width is thinner than 1280 say would be a good compromise. I know those auto layout algorithms can do funny things in edge cases.

I’m also trying to figure out what pc window size is equivalent.. i had thought its like 1024x768 with high dot pitch (2048x1500) but it might be more like a 1280x800 window.. I’ll take some comparison screenshots using the top menu ..

**Comment by dobkeratops** [comment]
Right this appears to confirm: the iPad browser seems equivalent to a 1280x800 pixel window
![B6A7E56B-4661-4B44-8F11-0C0B330DE4A0](https://user-images.githubusercontent.com/1120754/115107315-fcfcc480-9f61-11eb-98f9-6b69480981d0.jpeg)

**Comment by dobkeratops** [comment]
Regarding iPad pen text entry support.. one possible piece of streamlining would be to turn the label to lowercase *if it matches one in the official list* (eg “Car”=“car”)  but you might still want to enter capitals for product codes, acronyms, place names for free labels, so autoconverting all the time might backfire .

**Comment by bbernhard** [comment]
I just went down that rabbit hole and looked about an hour for a reliable way to detect whether a device is a mobile/tablet device. But with all those hybrid devices (desktop devices that have touch displays) and tablets with a really high screen resolution it's not reliably possible anymore to detect the device type. Oh man, I wish back those days where that information could be extracted from the user agent string...

What I've found is that media queries is probably the most reliable way right now to check whether a device is a mobile/tablet. So I guess I'll use some arbitrary screen width (like 1028px) as a threshold. Everything below that will be classified as mobile/tablet device and everything above that as desktop. That's certainly not perfect, but I guess it could work (maybe we need to tweak the threshold a bit in case it doesn't work for your device). I'll later do some alpha testing on my tablet and if everything works I think I could push it to production some time tomorrow.

**Comment by bbernhard** [comment]
> Regarding iPad pen text entry support.. one possible piece of streamlining would be to turn the label to lowercase _if it matches one in the official list_ (eg “Car”=“car”) but you might still want to enter capitals for product codes, acronyms, place names for free labels, so autoconverting all the time might backfire .

Looks like there are maybe some ways to turn that off. https://stackoverflow.com/questions/25786916/android-keyboard-defaults-to-capitalized-first-for-html-input

I'll have to check that later :)

**Comment by dobkeratops** [comment]
> What I've found is that media queries is probably the most reliable way right now to check whether a device is a mobile/tablet. So I guess I'll use some arbitrary screen width (like 1028px)

right I think that will work, and it might even make sense on desktop PCs aswell. The specific size of the iPad does appear to logically be 1280 pixels wide by experiments above. (Evidently there’s some scaling, not just 2x pixels ie 1024 - 2048 as I originally thought)

**Comment by bbernhard** [comment]
I just pushed a new version to production. It contains the following changes:

* the properties toolbox should be hidden for devices with a screen width <= 1280px
* I've added `autocapitalize="none"` to some input fields - so mobile/tablet devices shouldn't capitalize the first letter anymore. Not sure though, if I've caught all input fields. So if you stumble across one where it isn't working, please let me know. 
* There's now a new query option `image.num_annotations` (e.g: `image.num_annotations = 1`). In theory it should also work with the `<` and `>` operators, but there are again some performance bottlenecks causing the browser to freeze - I really need to rework the database queries at some point to fix those issues.

**Comment by dobkeratops** [comment]
That’s great , I’ll try it out. Every one of those tweaks opens up some work or streamlines something

**Comment by dobkeratops** [comment]
Right, all helpful improvements.
Num annotations=0 is as good as I’d hoped it would be .. plenty of varied images with labels already added (the gap in the “annotations vs labels” graphs)
Non capitalised label entry using iPad handwriting means it’s much easier to work pure iPad+pen ... very relaxed experience
Removing the properties view in thin windows has indeed allowed it to use the full width for the work area. The pc experience is undamaged 
![034ABF4E-9DF5-4F41-ACBD-6DE40FC9781F](https://user-images.githubusercontent.com/1120754/115141906-b6c56500-a036-11eb-811e-e7cb13d06f06.jpeg)

eventually I’m sure the iPad view could be tweaked with pop ups/expandable UI to show the properties aswell.. but right now I just use labels, so I don’t need it (I can just submit graph nodes to document free labels , to turn them into equivalent properties)
